subsys/llext/llext.c

```diff
@@ -290,12 +297,16 @@ static int llext_map_sections(struct llext_loader *ldr, struct llext *ext)
 		ldr->sect_map[i] = mem_idx;
 	}
 
-	ldr->prog_data_size = high_shdr.sh_size + high_shdr.sh_offset - low_shdr.sh_offset;
+	size_t prog_data_size = data_shdr_high.sh_size +
```

Comment:
We could just as well calculate the total size of these sections in the loop no?

Reply:
At present after the loop we use the low read-only and writable section headers as "templates" for a "cumulative" header, not just the size https://github.com/zephyrproject-rtos/zephyr/pull/73531/commits/cb2d34cfc9a6c167eef3e48b493ada43af55cd06#diff-af3b33d1fd9f53bfe92ba88f7a3941228401d26b9b9e44e80ff19f38133c17fdR306-R309